### DISCUSSION TITLE
Sleep for 12 hour

### DISCUSSION THREAD

**Discussion by CrabbyPete** [micropython/micropython#13252] [RP2040 / Pico]
I have a Raspberry Pi Pico which is powered by battery. Its monitoring an alarm, but I want it to sleep and wake up once or twice a day, and I need to go into low power sleep. machine.lightsleep only works for 72 minutes. Is there a way to put it to sleep for long periods of time?

**Comment by robert-hh** [comment]
The rp2 uses a 32bit counter @ 1MHz for lightsleep. That limits the time to about 72 minutes. If you need longer times, you can make a wrapper which uses several lightsleep periods.

**Reply by CrabbyPete** [reply]
If I do a machine.lightsleep() with no arguments can I get the code to wake up if something is written to the UART?

**Reply by KatDrone** [reply]
This maybe a little late, but posting anyway for others who may have the same problem - 

The machine.deep/light sleep problem has been an ongoing frustration / limitation with "stable" versions of 
micropython on RP2040. 

In late 2021 ghubcoder released a fork of micropython that fixed the deepsleep problem. See 
https://ghubcoder.github.io/posts/deep-sleeping-the-pico-micropython/#release
I have been using it for the last 2.5 yrs without any problem. It is not limited to 72 minutes. I have had it consistently
sleeping for 12 + hours months at a time without any problems. Because the build is quite old (at this point) there isn't support for  Wifi and I haven't been able to get ulab successfully compiled into it. It does however support everything else - UART, I2C, onewire etc. 
The RTC in sleep mode consistently maintains accuracy. All in all an excellent stable build as long as wifi and 
numpy are not necessary. Not sure why the code wasn't merged with the main code base ...

**Reply by scruss** [reply]
> Not sure why the code wasn't merged with the main code base ...

Was it ever submitted as a PR?
